### PR TITLE
Match the generator device to the pipeline for DDPM and DDIM

### PR DESCRIPTION
**Pull request by anton-l**
Alternative to https://github.com/huggingface/diffusers/pull/1215

Generates the initial and variance noise for `DDPM` and `DDIM` on the same device as the models. 
This is a breaking change in terms of reproducibility, but makes the pipelines behave the same way as `StableDiffusion`

### CODE REVIEW

File: src/diffusers/pipelines/ddim/pipeline_ddim.py [hunk 1]
```diff
@@ -75,32 +75,39 @@ def __call__(
...
             (batch_size, self.unet.in_channels, self.unet.sample_size, self.unet.sample_size),
             generator=generator,
+            device=self.device,
         )
-        image = image.to(self.device)
 
         # set step values
         self.scheduler.set_timesteps(num_inference_steps)
 
-        # Ignore use_clipped_model_output if the scheduler doesn't accept this argument
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/ddim/pipeline_ddim.py:88]: why remove this? Is this related to the PR?

**Review by patrickvonplaten — APPROVED** [review]
Great this looks perfect to me - thanks a lot for also adding backwards compatibility here. Very nice!

**Comment by patrickvonplaten** [comment]
I prefer this over #1215 - wdyt @pcuenca @patil-suraj @anton-l ?

**Review by pcuenca — APPROVED** [review]
Looks good to me!

**Comment by pcuenca** [comment]
I think this solution is clearer for the user and more consistent. The Generator with device pattern is already present in some of the READMEs, maybe it's worth adding a small section to the general README or the docs.
